invoices/admin.py

```diff
@@ -1,6 +1,33 @@
+from django.utils.translation import ugettext_lazy as _
+from django.http import HttpResponse
 from django.contrib import admin
 
-from .models import Sender
+from .models import Sender, Address, Invoice
+from .utils import zip_files
+
+
+def invoice_export_to_xml(modeladmin, request, queryset):
+    if len(queryset) == 1:
+        model = queryset[0]
+        response = HttpResponse(model.to_xml(), content_type='text/xml')
+        response['Content-Disposition'] = f'attachment; filename={model.get_filename()}'
+        return response
+
+    files = []
+    for model in queryset:
+        files.append((model.get_filename(), model.to_xml()))
+    response = HttpResponse(zip_files(files), content_type='application/zip')
+    response['Content-Disposition'] = 'attachment; filename=invoices.zip'
```

Comment:
I will have a better look soon-ish (I'm creating the project locally) but usually the browser isn't able to say the remaining time/the size of the download if you don't set the `Content-Length` header (with the bytes of the zip or single file).

Before doing anything I want to test just in case the browser somehow is able to understand it w/o `Content-Length` :)

It might not matter here because the files will be very small anyway, so It's more of a nice to have than anything else, you can ignore this comment :)

Reply:
That's a good idea!